src/internal/modules/charge-information/forms/charge-element/quantities.js

```diff
@@ -58,7 +58,7 @@ const form = request => {
 };
 
 const schema = (request) => {
-  const nonZeroNumberWithWSixDpRegex = /^[1-9]\d*\.?\d{0,6}$/;
+  const nonZeroNumberWithWSixDpRegex = /^\s*(?=.*[1-9])\d*(?:\.\d{1,6})?\s*$/;
```

Comment:
I don't 100% follow the regex but I have tested it and it appears to fix the issues described.
We could consider removing the `\s*` here, and adding `.trim()` in the Joi validator before the regex.  This should trim the whitespace before it reaches the regex.